test/Utils.test.js

```diff
@@ -0,0 +1,64 @@
+/* global describe it web3 */
+const assert = require('chai').use(require('chai-as-promised')).assert
+const sigUtil = require('eth-sig-util')
+
+const getDataToSign = require('../src/js/relayclient/EIP712/Eip712Helper')
+const Utils = require('../src/js/relayclient/utils')
+
+const EIP712Sig = artifacts.require('./EIP712Sig.sol')
+
+contract('Utils', async function (accounts) {
+  describe('#getEip712Signature()', async function () {
+    it('shuld generate a valid EIP-712 compatible signature', async function () {
+      const senderAccount = accounts[0]
+      const senderNonce = 5
+      const target = accounts[5]
+      const encodedFunction = '0xdeadbeef'
+      const pctRelayFee = 15
+      const gasPrice = 10000000
+      const gasLimit = 500000
+      const relayHub = accounts[8]
+      const relayAddress = accounts[9]
+
+      const sig = await Utils.getEip712Signature({
+        web3,
+        senderAccount,
+        senderNonce,
+        target,
+        encodedFunction,
+        pctRelayFee,
+        gasPrice,
+        gasLimit,
+        relayHub,
+        relayAddress
+      })
+
+      const data = await getDataToSign({
+        web3,
+        senderAccount,
+        senderNonce,
+        target,
+        encodedFunction,
+        pctRelayFee,
+        gasPrice,
+        gasLimit,
+        relayHub,
+        relayAddress
+      })
+      const recoveredAccount = sigUtil.recoverTypedSignature_v4({
```

Comment:
I wonder: does this test also fails on inner struct? 
there's a reference method in the EIP: https://github.com/ethereum/EIPs/blob/master/assets/eip-712/Example.js
does it give different result?

Reply:
I can try tomorrow, but I want to finish the task first

Reply:
did you check that?

Reply:
I fixed the inner struct signing, the comment is not relevant. Anyways, it does not fail.